python/mxnet/optimizer.py

```diff
@@ -80,6 +71,20 @@ def set_lr_scale(self, args_lrscale):
         """
         self.lr_scale = args_lrscale.copy()
 
+    def _update_count(self, index):
+        """
+        update num_update
+
+        Parameters:
+        index : int
+            The index will be updated
+        """
+        if index not in self._index_update_count:
+            self._index_update_count[index] = 0
+        self._index_update_count[index] += 1
+        self.num_update = max(self._index_update_count[index], self.num_update)
```

Comment:
both the optimizer and the lr_scheduler run on the server nodes for distributed training. the only inconsistency happens on the asgd setting that different servers may have different update_count, but the difference is upper bounded.